### PR TITLE
tests: run modified tests with full coverage

### PR DESCRIPTION
**Pull request by nashif**
This addition would catch tests being added or modified and would run those tests with --all and catch issues before they get merged.

### CODE REVIEW

File: scripts/ci/get_modified_tests.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# A script to check email identities in commit message. We verify the author
```
Comment (andrewboie) [inline scripts/ci/get_modified_tests.py]: is that what this script does?

File: scripts/ci/get_modified_tests.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    format = logging.Formatter('%(levelname)-8s: %(message)s')
+    console.setFormatter(format)
+    logger = logging.getLogger('')
+    logger.addHandler(console)
+    logger.setLevel(level)
+
+    logging.debug("Log init completed")
+
+def parse_args():
+    parser = argparse.ArgumentParser(
```
Comment (andrewboie) [inline scripts/ci/get_modified_tests.py]: was this copied from some other script?

File: .shippable.yml [hunk 3]
```diff
@@ -82,18 +90,10 @@ build:
             cp ./scripts/sanity_chk/last_sanity.xml shippable/testresults/;
             aws s3 cp ./scripts/sanity_chk/last_sanity.xml ${S3_PATH}/sanitycheck.xml;
           fi;
-    on_failure:
```
Comment (andrewboie) [inline .shippable.yml]: did we want to remove this logic?

Comment (nashif) [inline .shippable.yml]: yeah, both on_success and on_failure had exactly the same content, so there was duplication.

File: scripts/ci/get_modified_tests.py (hunk 2 above)
Comment (nashif) [inline scripts/ci/get_modified_tests.py]: oops^2, used an existing script we already had.

File: scripts/ci/get_modified_tests.py (hunk 1 above)
Comment (nashif) [inline scripts/ci/get_modified_tests.py]: oops

**Comment by nashif** [comment]
@andrewboie updated

File: .shippable.yml [hunk 4]
```diff
@@ -65,11 +66,18 @@ build:
             ./scripts/ci/check-compliance.py --commits ${COMMIT_RANGE} || true;
           fi;
       - >
+          python2 ./scripts/ci/get_modified_tests.py --commits origin/${PULL_REQUEST_BASE_BRANCH}..HEAD;
```
Comment (inakypg) [inline .shippable.yml]: suggest having the modified_test.args file being printed on the stdout of get_modified_tests to have clarity on where it comes from:
python2 ./scripts/ci/get_modified_tests.py ... > modified_tests.args
if [ -s modified_tests.args ]; then
   blah blah
fi

Comment (inakypg) [inline .shippable.yml]: why python2? the script has a python3 #!

Comment (nashif) [inline .shippable.yml]: sounds good

Comment (nashif) [inline .shippable.yml]: realised I am missing sh for python3, need to update the docker image at some point, will add a FIXME in there.

File: scripts/ci/get_modified_tests.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# A script to generate a list of tests that have changed or added and create an
+# arguemnts file for sanitycheck to allow running those tests with --all
+
```
Comment (inakypg) [inline scripts/ci/get_modified_tests.py]: So what is the logic on this? it seems to me that only if there is a change in (testcase|sample).yaml it is caught.

I'd start with "any changes in samples/ or tests/ have to be looked at" -- now . I'd suggest if the path matches (so a leaf file):

"(samples|tests)/.+/.+/[^/]+"

then the path has to be added to a set (not a list), as multiple matches (let's say in the Makefile or a source file etc) wouldn't have to result in multiple additions.

Comment (nashif) [inline scripts/ci/get_modified_tests.py]: thats what the script does. It looks at any changed files, then walks up the tree and looks for yaml files, if it finds any, it adds to the set.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by inakypg** [comment]
LGTM

**Pull request merged**